dbt_platform_helper/providers/files.py

```diff
@@ -8,26 +7,21 @@ def load(path: str) -> str:
         pass
 
     @staticmethod
-    def mkfile(base_path: str, file_path: str, contents: str, overwrite=False) -> str:
-        file_path = Path(file_path)
-        file = Path(base_path).joinpath(file_path)
-        file_exists = file.exists()
-
-        if not file_path.parent.exists():
-            makedirs(file_path.parent)
-
+    def mkfile(base_path: str, file_name: str, contents: str, overwrite=False) -> str:
+        file_path = Path(base_path).joinpath(file_name)
+        file_exists = file_path.exists()
         if file_exists and not overwrite:
             return f"File {file_path} exists; doing nothing"
 
-        action = "overwritten" if file_exists and overwrite else "created"
+        file_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
Before, the mkdirs was not taking into account the base_dir and so if base_dir was just "." for example it would work fine. If it was an actual directory, it would fail.